src/main/java/org/opensrp/web/controller/ClientMigrationFileResource.java

```diff
@@ -0,0 +1,165 @@
+
+    public static final String MIGRATION_FILENAME_PATTERN = "(\\d).(up|down).sql";
+
+    @Autowired
+    public void setObjectMapper(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    @Autowired
+    public void setClientMigrationFileService(ClientMigrationFileService clientMigrationFileService) {
+        this.clientMigrationFileService = clientMigrationFileService;
+    }
+
+    @RequestMapping(method = RequestMethod.POST, consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
+    public ResponseEntity<String> create(@RequestParam(value = "identifier", required = false) String finalIdentifier,
+            @RequestParam(value = "filename", required = false) String finalFilename,
+            @RequestParam(value = "jurisdiction", required = false) String jurisdiction,
+            @RequestParam(value = "version") int version,
+            @RequestParam("migration_file") MultipartFile migrationFile) {
+        if (migrationFile.isEmpty()) {
+            return new ResponseEntity<>("Migration file is empty/missing", HttpStatus.BAD_REQUEST);
+        }
+
+        if (!migrationFile.getOriginalFilename().matches(MIGRATION_FILENAME_PATTERN)) {
+            return new ResponseEntity<>("The migration filename does not obey the pattern " + MIGRATION_FILENAME_PATTERN, HttpStatus.BAD_REQUEST);
+        }
+
+
+        byte[] bytes;
+        try {
+            bytes = migrationFile.getBytes();
+        } catch (IOException e) {
+            logger.error("Error occurred trying to read uploaded file", e);
+            return new ResponseEntity<>("Invalid file", HttpStatus.BAD_REQUEST);
+        }
+
+        // The file storage can be easily switched here -> Kindly implement using a storage contracts if you do so
+        String filename = finalFilename != null ? finalFilename : migrationFile.getOriginalFilename();
+        String identifier = finalIdentifier != null ? finalIdentifier : filename;
+
+        String fileContentString = new String(bytes);
+
+        ClientMigrationFile clientMigrationFile = new ClientMigrationFile();
+        clientMigrationFile.setIdentifier(identifier);
+        clientMigrationFile.setFilename(filename);
+        clientMigrationFile.setVersion(version);
+        clientMigrationFile.setCreatedAt(new Date());
+        clientMigrationFile.setFileContents(fileContentString);
+        clientMigrationFile.setJurisdiction(jurisdiction);
+        clientMigrationFile.setOnObjectStorage(false);
+
+        // TODO: This should be handled on the manifest upload
+        //clientMigrationFile.setManifestId(4);
+
+        clientMigrationFileService.addClientMigrationFile(clientMigrationFile);
+        return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, consumes = { MediaType.MULTIPART_FORM_DATA_VALUE })
+    public ResponseEntity<String> update(@RequestParam(value = "identifier", required = false) String finalIdentifier,
```

Comment:
This method looks to share 80% of the code with the POST request, we could refactor both to share a single function